packages/opentelemetry-semantic-conventions/src/resource/SemanticResourceAttributes.ts

```diff
@@ -14,155 +14,837 @@
+  NAME +
+  SPACE) as 'service.namespace';
+
+/**
+ * The string ID of the service instance.
+ *
+ * Note: MUST be unique for each instance of the same `service.namespace,service.name` pair (in other words `service.namespace,service.name,service.instance.id` triplet MUST be globally unique). The ID helps to distinguish instances of the same service that exist at the same time (e.g. instances of a horizontally scaled service). It is preferable for the ID to be persistent and stay the same for the lifetime of the service instance, however it is acceptable that the ID is ephemeral and changes during important lifetime events for the service (e.g. service restarts). If the service has no inherent unique ID that can be used as the value of this attribute it is recommended to generate a random Version 1 or Version 4 RFC 4122 UUID (services aiming for reproducible UUIDs may also use Version 5, see RFC 4122 for more recommendations).
+ */
+export const SEMRESATTRS_SERVICE_INSTANCE_ID = (SERVICE_DOT +
+  INSTANCE +
+  DOT +
+  ID) as 'service.instance.id';
+
+/**
+ * The version string of the service API or implementation.
+ */
+export const SEMRESATTRS_SERVICE_VERSION = (SERVICE_DOT +
+  VERSION) as 'service.version';
+
+/**
+ * The name of the telemetry SDK as defined above.
+ */
+export const SEMRESATTRS_TELEMETRY_SDK_NAME = (TELEMETRY_DOT +
+  SDK_DOT +
+  NAME) as 'telemetry.sdk.name';
+
+/**
+ * The language of the telemetry SDK.
+ */
+export const SEMRESATTRS_TELEMETRY_SDK_LANGUAGE = (TELEMETRY_DOT +
+  SDK_DOT +
+  'language') as 'telemetry.sdk.language';
+
+/**
+ * The version string of the telemetry SDK.
+ */
+export const SEMRESATTRS_TELEMETRY_SDK_VERSION = (TELEMETRY_DOT +
+  SDK_DOT +
+  VERSION) as 'telemetry.sdk.version';
+
+/**
+ * The version string of the auto instrumentation agent, if used.
+ */
+export const SEMRESATTRS_TELEMETRY_AUTO_VERSION = (TELEMETRY_DOT +
+  'auto.version') as 'telemetry.auto.version';
+
+/**
+ * The name of the web engine.
+ */
+export const SEMRESATTRS_WEBENGINE_NAME = (WEBENGINE_DOT +
+  NAME) as 'webengine.name';
+
+/**
+ * The version of the web engine.
+ */
+export const SEMRESATTRS_WEBENGINE_VERSION = (WEBENGINE_DOT +
+  VERSION) as 'webengine.version';
+
+/**
+ * Additional description of the web engine (e.g. detailed version and edition information).
+ */
+export const SEMRESATTRS_WEBENGINE_DESCRIPTION = (WEBENGINE_DOT +
+  DESCRIPTION) as 'webengine.description';
+
+/**
+ * Definition of available values for SemanticResourceAttributes
+ * This type is used for backward compatibility, you should use the individual exported
+ * constants SemanticResourceAttributes_XXXXX rather than the exported constant map. As any single reference
+ * to a constant map value will result in all strings being included into your bundle.
+ * @deprecated Use the SEMRESATTRS_XXXXX constants rather than the SemanticResourceAttributes.XXXXX for bundle minification.
+ */
+export type SemanticResourceAttributes = {
```

Comment:
Creating a strong `type` definition of the existing class to make sure that everything is correctly defined and passed, without this strong type it would actually be possible to pass the wrong (or missing) strings to the namespace object construction on line 1164.

Now if you add or remove something that is expected TypeScript is very unhappy.